sdk/src/main/java/io/customer/sdk/data/communication/CustomerIOUrlHandler.kt

```diff
@@ -1,13 +1,27 @@
 package io.customer.sdk.data.communication
 
-import android.net.Uri
+import android.content.Intent
+import androidx.core.app.TaskStackBuilder
 
 interface CustomerIOUrlHandler {
 
     /**
      * Callback called when deeplink action is performed.
-     * @param uri Deeplink URL
-     * @return Boolean return TRUE, if the URI was handled otherwise false.
+     *
+     * @param url link received in notification data
+     * @return intent to launch on notification click; null to let the SDK handle this
      */
-    fun handleCustomerIOUrl(uri: Uri): Boolean
+    fun createIntentForLink(url: String?): Intent?
```

Comment:
This function's job is for customers to implement in their app to be able to return back to the SDK an `Intent` to start when a push is clicked. 

However, the function name and parameters `create...ForLink()` and `url: String?` make it seem like customers should only implement this function if they want to handle a deep link when a push is pressed. 

I would like to suggest that we instead change this to: 

```suggestion
    fun createIntentForPush(parsedPush: CustomerIOParsedPushPayload): Intent?
```

where `CustomerIOParsedPushPayload` is an object that contains properties inside for the parsed push like this:
```kotlin
val push: CustomerIOParsedPushPayload = ...
push.deepLink: String? 
push.cioDeliveryId: String 
push.title: String 
push.body: String 
```

This is how to do "deep link handling" on the iOS SDK now. Instead of allowing customers to handle deep links in a push, we give them a parsed push object and allow them to handle a push when it's clicked, not just when it's a deep link that's pushed.